scripts/core/menu/menu.ts

```diff
@@ -91,6 +104,71 @@ angular.module('superdesk.core.menu', [
                     scope.isTestEnvironment = config.isTestEnvironment;
                     scope.environmentName = config.environmentName;
 
+                    // menu items and groups - start
+                    let group = null;
+
+                    scope.items = [];
+
+                    workspaceMenu
+                        .filter((item) => !item.if || scope.$eval(item.if))
+                        .forEach((item) => {
+                            const itemGroup = item.group || group;
+
+                            if (itemGroup !== group) {
+                                scope.items.push({hr: 1});
+                                group = itemGroup;
+                            }
+
+                            scope.items.push(item);
+                        });
+                    // menu items and groups - end
+
+                    /*
+                        Marking item as active even if current path doesn't match exactly to item href path
+                        For example, if current path is /a/b/c and there is only menu item for /a/b,
+                        /a/b should be set active.
+                    */
+                    function getActiveMenuItemPath(currentPath: string) {
```

Comment:
imo you're looking for longest matching prefix in strings, so this could be something like:
```
scope.items
    .filter((item) => typeof item.href === 'string' && item.href.length > 0 && currentPath.startsWith(item.href))
    .map((item) => item.href)
    .reduce((longest, current) => longest == null || longest.length < current.length ? current : longest);
```
which imo makes it more obvious and wound't need all those comments.

Reply:
:+1: Great suggestion. I only changed it to compare the number of occurrences of `/` instead of character count, since `a/b/c/d/e/f` is deeper even if `a/b/c/ddddddddddd` is longer by the number of characters.

Reply:
but they wouldn't match both as a prefix if they are different, would they?

Reply:
so at the end one must be prefix of another if they are both a prefix of current url, and only thing that matters is which one is longer imo

Reply:
You're right. I've removed the splitting part.